plugins/outputs/pulsar/pulsar.go

```diff
@@ -0,0 +1,378 @@
+ # compression_type = 0
+
+ ## MaxPendingMessages set the max size of the queue holding the messages pending to receive an
+ ## acknowledgment from the broker.
+ # max_pending_messages = 1000
+
+ ## BatchingMaxMessages set the maximum number of messages permitted in a batch. (default: metric_batch_size)
+ ## If set to a value greater than 1, messages will be queued until this threshold is reached or
+ ## batch interval has elapsed. By Default it is set to "metric_batch_size"
+ # batching_max_messages = 1000
+
+ ## HashingScheme change the "HashingScheme" used to chose the partition on where to publish a particular message.
+ ##  JavaStringHash : JavaStringHash Hshing
+ ##  Murmur3_32Hash : Murmur3_32Hash Hashing
+ # hashing_scheme = "JavaStringHash"
+
+
+ ## Data format to output.
+ ## Each data format has its own unique set of configuration options, read
+ ## more about them here:
+ ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md
+ # data_format = "influx"
+
+`
+
+func ValidateTopicSuffixMethod(method string) error {
+	for _, validMethod := range ValidTopicSuffixMethods {
+		if method == validMethod {
+			return nil
+		}
+	}
+	return fmt.Errorf("Unknown topic suffix method provided: %s", method)
+}
+
+func (p *Pulsar) GetTopicName(metric telegraf.Metric) (telegraf.Metric, string) {
+	topic := p.Topic
+	if p.TopicTag != "" {
+		if t, ok := metric.GetTag(p.TopicTag); ok {
+			topic = p.tenantNameSpace + "/" + t
+			// If excluding the topic tag, a copy is required to avoid modifying
+			// the metric buffer.
+			if p.ExcludeTopicTag {
+				metric = metric.Copy()
+				metric.Accept()
+				metric.RemoveTag(p.TopicTag)
+			}
+		}
+	}
+	return metric, topic
+}
+
+func (p *Pulsar) GetProducer(topic string, metricBatchSize uint) (pulsar.Producer, error) {
+	if producer, ok := p.producerCache[topic]; !ok {
+		producerOptions := pulsar.ProducerOptions{
+			Topic:               topic,
+			BatchingMaxMessages: metricBatchSize,
+		}
+		switch p.CompressionType {
+		case 0:
+			producerOptions.CompressionType = pulsar.NoCompression
+		case 1:
+			producerOptions.CompressionType = pulsar.LZ4
+		case 2:
+			producerOptions.CompressionType = pulsar.ZLib
+		case 4:
+			producerOptions.CompressionType = pulsar.ZSTD
+		default:
+			producerOptions.CompressionType = pulsar.NoCompression
+		}
+
+		switch p.HashingScheme {
+		case "Murmur3_32Hash":
+			producerOptions.HashingScheme = pulsar.Murmur3_32Hash
+		case "JavaStringHash":
+			producerOptions.HashingScheme = pulsar.JavaStringHash
+		default:
+			producerOptions.HashingScheme = pulsar.JavaStringHash
+		}
+		if p.MaxPendingMessages > 0 {
+			producerOptions.MaxPendingMessages = p.MaxPendingMessages
+		}
+		duration, err := time.ParseDuration(p.BatchingMaxPublishDelay)
+		if err != nil {
+			producerOptions.BatchingMaxPublishDelay = duration
+		}
+		producer, err := p.client.CreateProducer(producerOptions)
+		if err != nil {
+			return nil, err
+		} else {
```

Comment:
No need for the `else` as you return in the `if`. This saves another indention level.